nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,294 @@
+        This value will be applied on the signals by passing to signal.clean
+        Please see the related documentation signal.clean for more details.
+        passed to NiftiMapsMasker.
+
+    t_r: float, default is None, optional
+        Repetition time in sec. This value is given to signal.clean
+        Please see the related documentation for details.
+        passed to NiftiMapsMasker.
+
+    memory: instance of joblib.Memory, string, default is None, optional
+        Used to cache the masking process. If a string is given, the path
+        is set with this string as a folder name in the directory.
+        passed to NiftiMapsMasker.
+
+    memory_level: int, default is 0, optional
+        Aggressiveness of memory catching. The higher the number, the higher
+        the number of functions that will be cached. Zero mean no caching.
+        passed to NiftiMapsMasker.
+
+    verbose: int, default is 0, optional
+        Indicates the level of verbosity by printing the message. Zero indicates
+        nothing is printed.
+
+    Attributes
+    ----------
+    regions_: Niimg-like image/object
+        a list of seperated regions with each region lying on a 3D volume
+        concatenated into a 4D Nifti like object.
+
+    index_: a numpy array
+        an array of list of indices where each index value is assigned to
+        each seperate region of its corresponding family of brain maps.
+
+    signals_: a numpy array
+        a list of averaged timeseries signals of the subjects extracted from
+        each region.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask_img=None, min_size=50,
+                 threshold="80%", thresholding_strategy='percentile',
+                 extractor='local_regions', peak_local_smooth=6.,
+                 standardize=False, detrend=False,
+                 low_pass=None, high_pass=None, t_r=None,
+                 memory=Memory(cachedir=None), memory_level=0, verbose=0):
+        super(RegionExtractor, self).__init__(
+            maps_img=maps_img, mask_img=mask_img,
+            standardize=standardize, detrend=detrend, low_pass=low_pass,
+            high_pass=high_pass, t_r=t_r, memory=memory,
+            memory_level=memory_level, verbose=verbose)
+        self.maps_img = maps_img
+        self.min_size = min_size
+        self.thresholding_strategy = thresholding_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+        self.peak_local_smooth = peak_local_smooth
+
+    def fit(self, X=None, y=None):
+        """ Prepare the data and setup for the region extraction
+
+        """
+        maps_img = check_niimg_4d(self.maps_img)
+
+        # foreground extraction
+        self.threshold_maps_img_ = threshold_img(
```

Comment:
Is there any reason to keep these maps? I think that you should just discard them after region extraction.